File: 05.11.20︱Metal Kirikot/Command.cs
using DSharpPlus;

namespace Metal_Kirikot.Utilities.Events {
    public class Command : DiscordEvent {
         public string[] Usage { get; set; }
         public CommandGroup CommandGroup { get; set; }
         public CommandType CommandType { get; set; }
         public Permissions[] RequiredPermissions { get; set; }
    }
}

File: OUDATED/NEWEST︱TypeScript/Source/Utilities/Command.ts
import { PermissionString } from 'discord.js'
import { EventArgs } from './EventArgs'

export class Command {
    public Options: CommandOptions;
    constructor(options: CommandOptions) {
        this.Options = options;
    }
}
export interface CommandOptions { 
    Name: string; 
    Usage: string[];
    CommandGroup: CommandGroup,
    CommandType: CommandType; 
    RequiredPermissions: PermissionString[]; 
    Execute: (eventArgs: EventArgs) => void;
}
export enum CommandGroup {
    Any
    /*Moderation,
    LevelUp,
    Color,
    Fun,
    Music*/
}
export enum CommandType {
    BotOwnerOnly,
    BetaTesterOnly,
    GuildOwnerOnly,
    GuildOnly,
    Everywhere
}